avro-serializer/src/main/java/io/confluent/kafka/serializers/AbstractKafkaAvroDeserializer.java

```diff
@@ -161,11 +164,20 @@ protected GenericContainerWithVersion deserializeWithSchemaAndVersion(
         return new GenericContainerWithVersion(new NonRecordContainer(schema.rawSchema(), result),
             version);
       }
-    } catch (RestClientException | IOException e) {
+    } catch (IOException e) {
       throw new SerializationException("Error retrieving Avro "
                                       + getSchemaType(isKey)
                                       + " schema version for id "
                                       + context.getSchemaId(), e);
+    } catch (RestClientException e) {
+      if (Family.familyOf(e.getErrorCode()) == Family.CLIENT_ERROR) {
+        throw new InvalidConfigurationException(e.getMessage());
+      } else {
+        throw new SerializationException("Error retrieving Avro "
+            + getSchemaType(isKey)
+            + " schema version for id "
+            + context.getSchemaId(), e);
+      }
```

Comment:
Let's move this logic to a protected static method called `toKafkaException` in `AbstractKafkaSchemaSerDe` and then replace this with `throw toKafkaException(e)`